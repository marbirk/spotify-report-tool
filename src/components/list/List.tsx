import React from 'react'

interface ListProps {
    children: React.ReactNode
    className?: string
}

const List = ({ children, ...props }: ListProps) => (
    <ul {...props}>{children}</ul>
)

export default List
