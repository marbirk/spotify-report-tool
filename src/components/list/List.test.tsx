import React from 'react'
import { render, screen } from '@testing-library/react'

import List from './List'

describe('List', () => {
    const entries = ['entry 1', 'entry 2', 'entry 3']
    test('renders correctly', () => {
        render(
            <List>
                {entries.map(item => (
                    <li key={item}>{item}</li>
                ))}
            </List>
        )
        const component = screen.getByText('entry 1')
        expect(component).toBeInTheDocument()
        expect(component.parentNode).toMatchSnapshot()
    })
})
