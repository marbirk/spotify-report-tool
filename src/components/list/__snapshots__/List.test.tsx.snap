// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<ul>
  <li>
    entry 1
  </li>
  <li>
    entry 2
  </li>
  <li>
    entry 3
  </li>
</ul>
`;
